include/onnxruntime/core/graph/graph.h

```diff
@@ -21,8 +21,19 @@
 #pragma warning(pop)
 #endif
 
+#if defined(__GNUC__)
+#pragma GCC diagnostic push
+
+#ifdef HAS_SHORTEN_64_TO_32
+#pragma GCC diagnostic ignored "-Wshorten-64-to-32"
+#endif
+
 #include "flatbuffers/flatbuffers.h"
 
+#if defined(__GNUC__)
+#pragma GCC diagnostic pop
+#endif
+
```

Comment:
if we need to generally suppress this warning in flatbuffers now, consider putting this in a separate wrapper header, e.g., `onnxruntime/core/common/flatbuffers.h`.

something like this:
https://github.com/microsoft/onnxruntime/blob/18ad8587a696ac6d7fd1e73d880247d2722755e1/onnxruntime/core/common/safeint.h#L28-L38

also, HAS_SHORTEN_64_TO_32 is defined in onnxruntime_config.h, so it would be good to explicitly include it